SingularityService/src/test/java/com/hubspot/singularity/mesos/SingularityMesosTaskBuilderTest.java

```diff
@@ -138,6 +141,40 @@ public void testJobUserPassedAsEnvironmentVariable() {
   }
 
   @Test
+  public void testEnvironmentVariableOverrides() {
+    Map<String, String> overrideVariables = new HashMap<>();
+    overrideVariables.put("HS_ACCESS_GROUP", "test");
+    overrideVariables.put("STARTED_BY_USER", "notTestUser");
+
+    SingularityRunNowRequest runNowRequest = new SingularityRunNowRequestBuilder()
+        .environmentVariables(overrideVariables)
+        .build();
+
+    final SingularityRequest request = new SingularityRequestBuilder("test", RequestType.WORKER)
+        .build();
+    final SingularityDeploy deploy = new SingularityDeployBuilder("test", "1")
+        .setCommand(Optional.of("/bin/echo hi"))
+        .setRunImmediately(Optional.of(runNowRequest))
+        .build();
+    final SingularityTaskRequest taskRequest = new SingularityTaskRequest(request, deploy, pendingTask);
```

Comment:
The pendingTask being used here is the one created in the `Before`. Since the environment overrides will be on the pending task (see more detail in task builder comments), we'll need to construct our own pending task to test this here